File: samples/sample-app-storage/src/main/java/com/azure/android/storage/sample/core/util/paging/PaginationOptions.java
package com.azure.android.storage.sample.core.util.paging;

import androidx.annotation.NonNull;
import androidx.paging.PagedList;

import java.util.concurrent.Executor;

public class PaginationOptions {
    private final Executor pageLoadExecutor;
    private PagedList.Config pagedListConfig;
    private boolean interactiveLoginEnabled;

    public PaginationOptions(Executor pageLoadExecutor) {
        this.pageLoadExecutor = pageLoadExecutor;
    }

    public Executor getPageLoadExecutor() {
        return this.pageLoadExecutor;
    }

    public PagedList.Config getPagedListConfig() {
        return this.pagedListConfig;
    }

    @NonNull
    public PaginationOptions setPagedListConfig(PagedList.Config config) {
        this.pagedListConfig = config;
        return this;
    }

    public boolean isInteractiveLoginEnabled() {
        return this.interactiveLoginEnabled;
    }

    public PaginationOptions enableInteractiveLogin(boolean enabled) {
        this.interactiveLoginEnabled = enabled;
        return this;
    }
}

File: samples/sample-app-storage-kotlin/src/main/java/com/azure/android/storage/sample/kotlin/core/util/paging/PaginationOptions.kt
package com.azure.android.storage.sample.kotlin.core.util.paging

import androidx.paging.PagedList
import java.util.concurrent.Executor

class PaginationOptions(val pageLoadExecutor: Executor) {
    var pagedListConfig: PagedList.Config? = null
        private set
    var isInteractiveLoginEnabled = false
        private set

    fun setPagedListConfig(config: PagedList.Config): PaginationOptions {
        pagedListConfig = config
        return this
    }

    fun enableInteractiveLogin(enabled: Boolean): PaginationOptions {
        isInteractiveLoginEnabled = enabled
        return this
    }
}
